### ISSUE TITLE
Status bar hover does not work well with magnifier

### ISSUE THREAD

**Issue by PurpleGranite** [microsoft/vscode#112220]
- VSCode Version:  1.51.1
- OS Version:  Windows 10, 1909

Steps to Reproduce:

1.  Be sight impaired
2.  Require a magnifier to be able to work
3.  Attempt to use VS Code
4.  VS Code generates a warning on the status bar
5.  Hover over the status bar, tooltip pops up
6.  Tooltip is significantly larger than the viewport from the magnifier, providing precisely zero useful information
7.  Tooltip disappears when attempting to screenshot, so cannot capture and read outwith code
8.  Tooltip disappears when attempting to move the magnified viewport to read the completely hidden parts of the tooltip
9.  Clicking the same area on the status bar gives a clearly different message, which is badly formatted

The 'pop-up' tooltips should be equally accessible as the rest of the application, and any message which appears on clisk should contain the same content, formatted in the same way as the tooltip.  This is not the case, and it's clear that this has never been tested by the team at any point.

Does this issue occur when all extensions are disabled?: Yes

**Comment by isidorn** [comment]
Thank you for providing feedback.
For our accessibility work we have been mostly focused on Screen Reader users and I do agree that we can improve the experience with Magnifiers. We plan to make a pass over a product regarding usage with a magnifier. Any further feedback would be appreciated.

As for this issue, the gist seems to be:
Status bar hover does not work well with magnifier. Thus renaming.

Since we are using the browser native hover, can you reproduce this issue if you just use Google Chrome and use a magnifier on a webpage that uses regular hovers? What is an example of an App that does a good job regarding natives hovers and the Windows magnifier?

**Comment by PurpleGranite** [comment]
The ironic thing is that _VS Code_ does hover well elsewhere, such as in IntelliSense popups - the absolutely critical point in making these things work well with magnification software is that you **must** be able to move the mouse pointer 'into' the popup and have that popup remain visible.

Additionally, it is very important that you do not confine your testing to the Windows built-in accessibility tools - anyone who actually requires accessibility tools such as magnifiers and screen readers for more than a few minutes at a time will tell you that the built in tools are nothing short of inadequate.  At a minimum, accessibility testing should be done with:

- Screen Readers
  - NVDA
  - JAWS
  - ZoomText Magnifier + Reader
- Magnifiers
  - Dolphin SuperNova
  - ZoomText

From what I can see, the popups from the status bar _aren't_ browser popups, they look very much like standard Windows OS tooltips - see the image below (taken with a smartphone, because, as described, getting a screenshot of this natively in Windows isn't possible!)

![image](https://user-images.githubusercontent.com/8159386/101928663-c8e94f80-3bcd-11eb-824c-b75935ea178a.png)

**Comment by isidorn** [comment]
Thanks for providing more details. For Screen Readers we test with NVDA, Orca on Linux and VoiceOver on mac.
Magnifiers we should give a test pass using the ones you listed, thanks for the suggestion.

As for the status hover: I believe on Windows we use the custom hover, thus also assigning this to @alexr00

**Comment by alexr00** [comment]
@bpasero do you own the status bar item? It looks like it's using the native hover, so forwarding on to you:

https://github.com/microsoft/vscode/blob/b07d19a768b42af2aa8f78f8b1b68978f14f9bd9/src/vs/workbench/browser/parts/statusbar/statusbarPart.ts#L826-L833

**Comment by bpasero** [comment]
Yeah, but what is the scope of this issue? If it is about native vs. custom hovers then surely I can find more locations in the UI that would not work well. Doesn't this mean we need to adopt the custom hover everywhere? Maybe only when accessibility mode is enabled?

**Comment by isidorn** [comment]
As far as I understood the issue it happens for all native hovers. However @PurpleGranite believes we are not using a native hover in the status bar, and it seems he was mistaken since we do use the native hover for status bar.

So yeah the request seems to be to use the custom hover in more places.
It would also be interesting to see what Chrome does with the native hover + magnifier and what are some of the apps that do this well.
